_docs/_latest/develop/dev/gcp-api.md

```diff
@@ -0,0 +1,220 @@
+clients, without having to rewrite these common methods for every API clients.
+
+You can [learn more about the mixin pattern here](https://www.ianlewis.org/en/mixins-and-python).
+
+You can see the currently available [mixins in Forseti here]({% link _docs/_latest/develop/reference/google.cloud.forseti.common.gcp_api.html %}).
+
+## Main Steps
+
+The main steps to add a new foo API client are:
+
+1. Define the API name & versions to be added.
+1. Create a new foo API client file.
+1. Create FooClient class to provide the entry point to the GCP API
+method. FooClient contains FooRepositoryClient.
+1. Create FooRepositoryClient class, and define the property method for the
+bar resource that you want to interact with the GCP API. FooRepositoryClient
+contains _FooBarRepository.
+1. Create a \_FooBarRepository class, to install the base GCP API
+functionalities (building requests, authentication, mixin classes).
+1. Use the new API client.
+
+A good example to look at, that is small, complete, and self-contained would be
+[cloudsql.py]({% link _docs/_latest/develop/reference/_modules/google/cloud/forseti/common/gcp_api/cloudsql.html %}).
+We will use it for the code-walk below.
+
+## Define the API name & versions to be added
+
+Add a new entry for the API and the versions that can be used in the
+`SUPPORTED_APIS` map.  You can see the available
+[the APIs provided by GCP](https://cloud.google.com/apis/docs/overview).
+
+[google/cloud/forseti/common/gcp_api/_supported_apis.py](https://github.com/GoogleCloudPlatform/forseti-security/blob/stable/google/cloud/forseti/common/gcp_api/_supported_apis.py)
+
+```python
+SUPPORTED_APIS = {
+    'bigquery': {
+        'default_version': 'v2',
+        'supported_versions': ['v2']
+    },
+    'cloudresourcemanager': {
+        'default_version': 'v1',
+        'supported_versions': ['v1', 'v2']
+    },
+    'foo': {
+        'default_version': 'v1',
+        'supported_versions': ['v1', 'beta']
+    },
+    'sqladmin': {
+        'default_version': 'v1beta4',
+        'supported_versions': ['v1beta4']
+    },
+}
+```
+
+## Create new foo API client file
+
+Create a new foo API client file called foo.py in
+[google/cloud/forseti/common/gcp_api]({% link _docs/_latest/develop/reference/google.cloud.forseti.common.gcp_api.html %})
+package.
+
+## Create FooClient class to provide the entry point to the GCP API method
+
+1. Create FooClient class in foo.py.
+1. Initialize with the allowed quota and FooRepositoryClient (see next section).
+1. Create a get_(repository_client_property)s e.g. `get_instances()` as below,
+if the repository client has property `instances`. In the example below,
+the repository has list mixin, which will invoke the actual GCP API call.
+
+[google/cloud/forseti/common/gcp_api/cloudsql.py]({% link _docs/_latest/develop/reference/_modules/google/cloud/forseti/common/gcp_api/cloudsql.html %})
+
+```python
+class CloudsqlClient(object):
```

Comment:
If possible, let's use the generated source to link to in /reference/.

Reply:
Please see my other comment here about having some code here in one place.